src/utils/timeTracker/index.js

```diff
@@ -0,0 +1,188 @@
+/**
+Copyright 2016 Split Software
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+**/
+
+'use strict';
+
+const uniqueId = require('lodash/uniqueId');
+const Logger = require('logplease');
+const timer = require('./timer');
+const thenable = require('../promise/thenable');
+
+// logger to be used on this module
+const logger = Logger.create('[TIME TRACKER]', {
+  showTimestamp: false,
+  showLevel: false,
+  useColors: false
+});
+// Map we will use for storing timers
+const timers = {};
```

Comment:
Yes, as we've spoken before. That's still pending to be solved, but it's not a new problem. I'll try to get a solution before merging this branch either way.